src/Tribe/PUE/Notices.php

```diff
@@ -87,26 +87,52 @@ public function register_name( $plugin_name ) {
 	/**
 	 * Restores plugins added on previous requests to the relevant notification
 	 * groups.
+	 *
+	 * @sice TBD Switched from array`array_merge_recursive` to `wp_parse_args` to fix an issue with data duplication.
 	 */
 	protected function populate() {
-		$this->saved_notices = (array) get_option( self::STORE_KEY, [] );
+		// Retrieve saved notices and ensure it's an array.
+		$saved_notices = get_option( self::STORE_KEY, [] );
 
-		if ( empty( $this->saved_notices ) ) {
-			return;
+		if ( ! is_array( $saved_notices ) ) {
+			$saved_notices = [];
+			update_option( self::STORE_KEY, $saved_notices );
 		}
 
-		$this->notices = array_merge_recursive( $this->notices, $this->saved_notices );
+		$this->saved_notices = $saved_notices;
+
+		// Sanitize and merge notices.
+		$this->notices = $this->sanitize_notices( wp_parse_args( $this->notices, $this->saved_notices ) );
```

Comment:
This becomes an issue you have to clear as soon as possible otherwise you risk memory exhaution